src/react/__tests__/useStore.test.tsx

```diff
@@ -90,6 +90,29 @@ describe('useStore', () => {
       ]
     `)
   })
+  it('should correct work, when store contains function', async () => {
+    const fn = jest.fn()
+    const changeStore = createEvent()
+    const $store = createStore(() => 0).on(changeStore, (_, p) => p)
+
+    const Display = () => {
+      const store = useStore($store)
+      fn()
```

Comment:
There should be a proof that issues does exists at all: your example [reveal](https://codesandbox.io/s/admiring-goldberg-i9q5l?file=/src/App.js) that current implementation works as expected

Reply:
[original issue](https://share.effector.dev/5T7AGosk) 
[new sandbox with proofs](https://codesandbox.io/s/sleepy-shadow-b9nf4?file=/src/App.js)